src/lib.rs

```diff
@@ -305,11 +330,40 @@ impl<T: AsRef<str>> fmt::Display for Grid<T> {
                 // We also only call `write_str` when we actually need padding as
                 // another optimization.
                 f.write_str(contents.as_ref())?;
-                if !last_in_row {
-                    if padding_size > 0 {
-                        f.write_str(&padding[0..padding_size])?;
+                // In case if this entry was the last on the current line,
+                // there is no need to check for separator or padding.
+                if !last_in_row && num + next < self.cells.len() {
```

Comment:
It might be nice to flip this condition and make it an early `continue` instead.